modules/buyerBidAdapter.js

```diff
@@ -0,0 +1,100 @@
+import * as utils from 'src/utils';
+import {config} from 'src/config';
+import {registerBidder} from 'src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'buyer';
+const ENDPOINT_URL = 'https://buyer.dspx.tv/request/';
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['buyer'],
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.placement);
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    return validBidRequests.map(bidRequest => {
+      const params = bidRequest.params;
+      const sizes = utils.parseSizesInput(bidRequest.sizes)[0];
+      const width = sizes.split('x')[0];
+      const height = sizes.split('x')[1];
+      const placementId = params.placement;
+
+      const rnd = Math.floor(Math.random() * 99999999999);
+      const referrer = encodeURIComponent(utils.getTopWindowUrl()); // bidderRequest.refererInfo ??
```

Comment:
I would suggest to use the `bidderRequest.referrerInfo.referer` field instead of the `utils.getTopWindowUrl()` as that will eventually be deprecated.